### PR TITLE
Add openstack_networking_secgroup_v2 datasource

### PR DESCRIPTION
**Pull request by mcanevet**

### CODE REVIEW

File: openstack/data_source_openstack_networking_secgroup_v2.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+	listOpts := groups.ListOpts{
+		Name:     d.Get("name").(string),
+		TenantID: d.Get("tenant_id").(string),
+	}
+
+	pages, err := groups.List(networkingClient, listOpts).AllPages()
+	allSecGroups, err := groups.ExtractGroups(pages)
+	if err != nil {
+		return fmt.Errorf("Unable to retrieve networks: %s", err)
```
Comment (fatmcgav) [inline openstack/data_source_openstack_networking_secgroup_v2.go]: Should this be `Unable to retrieve security groups:`?

File: openstack/data_source_openstack_networking_secgroup_v2.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	}
+
+	if len(allSecGroups) < 1 {
+		return fmt.Errorf("Your query returned no results. " +
+			"Please change your search criteria and try again.")
```
Comment (fatmcgav) [inline openstack/data_source_openstack_networking_secgroup_v2.go]: Is `No Security Group found with name: [name]` more accurate?

File: openstack/data_source_openstack_networking_secgroup_v2.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	}
+
+	if len(allSecGroups) > 1 {
+		return fmt.Errorf("Your query returned more than one result." +
+			" Please try a more specific search criteria")
```
Comment (fatmcgav) [inline openstack/data_source_openstack_networking_secgroup_v2.go]: Is `More than one Security Group found with name: [name]` more accurate?

**Comment by fatmcgav** [comment]
I think this will need another change in [`website/openstack.erb`](https://github.com/terraform-providers/terraform-provider-openstack/blob/master/website/openstack.erb) aswell - See https://github.com/terraform-providers/terraform-provider-openstack/commit/c9d5900d6e16c2890f8d5daaa9edb34d2b694268#diff-2bd5efeeeeef016e68bb6de1aea0f9a0 for an example...

**Comment by fatmcgav** [comment]
Also, should probably add an acceptance test for this if possible. https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/data_source_openstack_networking_network_v2_test.go should be a good base...

File: website/openstack.erb [hunk 4]
```diff
@@ -19,6 +19,9 @@
             <li<%= sidebar_current("docs-openstack-datasource-networking-network-v2") %>>
               <a href="/docs/providers/openstack/d/networking_network_v2.html">openstack_networking_network_v2</a>
             </li>
+            <li<%= sidebar_current("docs-openstack-datasource-networking-secgroup-v2") %>>
+              <a href="/docs/providers/openstack/d/networking_secgroup.html">openstack_networking_secgroup_v2</a>
```
Comment (fatmcgav) [inline website/openstack.erb]: I think this should be `networking_secgroup_v2.html`.

Comment (mcanevet) [inline website/openstack.erb]: indeed

**Comment by mcanevet** [comment]
@fatmcgav how can I run acceptance tests locally?

**Comment by fatmcgav** [comment]
@mcanevet I've run them in the past similar to this:
`TF_LOG=DEBUG make testacc TEST=./openstack TESTARGS="-run=TestAccComputeV2Flavor_basic"` 
Just replace the `TestAccComputeV2Flavor_basic` test-case with the one you're interested in. 

This assumes that your local env variables are set to enable access to Openstack. 

There is a packstack example in the repo aswell if you don't want to run against a 'real' Openstack cloud...

**Comment by mcanevet** [comment]
```
$ make testacc TEST=./openstack TESTARGS="-run=TestAccOpenStackNetworkingSecGroupV2DataSource_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./openstack -v -run=TestAccOpenStackNetworkingSecGroupV2DataSource_basic -timeout 120m
=== RUN   TestAccOpenStackNetworkingSecGroupV2DataSource_basic
--- PASS: TestAccOpenStackNetworkingSecGroupV2DataSource_basic (17.13s)
PASS
ok  	github.com/terraform-providers/terraform-provider-openstack/openstack	17.133s
```

**Comment by jtopjian** [comment]
I did a quick glance of this and it looks good. Nice, work @mcanevet!

I have one request if it's not too hard: add an option to search for a security group by ID. This was added to the [network data source](https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/data_source_openstack_networking_network_v2.go) shortly after it was added. This helps when the user wants to do ID-to-name queries, which can be especially useful in the `openstack_compute_instance_v2` resource where the security group accepts a name and not an ID.

**Comment by mcanevet** [comment]
@jtopjian done!

**Comment by fatmcgav** [comment]
@mcanevet Are you able to update the docs to include `secgroup_id` details?

**Comment by mcanevet** [comment]
@fatmcgav Done! I also added an acceptance test for `secgroup_id`

**Comment by fatmcgav** [comment]
@mcanevet Awesome, cheers for that. 

Will get it merged in this morning.

**Comment by fatmcgav** [comment]
Merged!

Cheers again for the contribution @mcanevet

**Pull request merged**